### DISCUSSION TITLE
ggml VS ncnn

### DISCUSSION THREAD

**Discussion by khengari77** [ggml-org/ggml#148] [General]
How does this library compare to [ncnn](https://github.com/Tencent/ncnn) ? I know that this library is relatively new but what pros and cons does it have in comparison to ncnn and is it from the community to come out with a library that competes with ncnn?

**Comment by appvoid** [comment]
They are similar in that:

- They both are difficult to modify for beginners.
- They both support multiplatform binaries.
- They look for a minimal C++ approach. (well, ncnn tries it)

The difference is **huge** though.

- This repo contains code maintained by Georgi and its amazing community. ncnn is maintained by Tencent.
- ggml is focused on large language models, ncnn is focused on computer vision.
- ggml philosophy is experimentation and "proved to work" mindset, which means less features and more stability. ncnn is more likely to have more features than you need, but that means that you are not on your own making the wheel.

> Edit: Forgot to add this last one about inference.

- ggml is highly focused on desktop inference, specially macOS.
- ncnn loves mobile platforms the most.


Hope it helps to give a better overview.
